deploy/buildtools/buildfarm.py

```diff
@@ -0,0 +1,356 @@
+                # add element w/ defaults
+
+                ip_addr = build_farm_host
+                dest_build_dir = default_build_dir
+            else:
+                raise Exception(f"""Unexpected YAML type provided in "{build_farm_hosts_key}" list. Must be dict or str.""")
+
+            if not dest_build_dir:
+                raise Exception("ERROR: Invalid null build dir")
+
+            self.build_hosts.append(BuildHost(ip_address=ip_addr, dest_build_dir=dest_build_dir))
+
+    def request_build_host(self, build_config: BuildConfig) -> None:
+        """Request build host to use for build config. Just assigns build config to build host since IP address
+        is already granted by something outside of FireSim."
+
+        Args:
+            build_config: Build config to request build host for.
+        """
+
+        if len(self.build_hosts) > self.build_hosts_allocated:
+            self.build_hosts[self.build_hosts_allocated].build_config = build_config
+            self.build_hosts_allocated += 1
+        else:
+            bcf = build_config.build_config_file
+            error_msg = f"ERROR: {bcf.num_builds} builds requested in `config_build.yaml` but {self.NAME()} build farm only provides {len(self.build_hosts)} build hosts (i.e. IPs)."
+            rootLogger.critical(error_msg)
+            raise Exception(error_msg)
+
+        return
+
+    def wait_on_build_host_initialization(self, build_config: BuildConfig) -> None:
+        """Nothing happens since the provided IP address is already granted by something outside FireSim.
+
+        Args:
+            build_config: Build config used to find build host that must ready.
+        """
+        return
+
+    def release_build_host(self, build_config: BuildConfig) -> None:
+        """ Nothing happens. Up to the IP address provider to cleanup after itself.
+
+        Args:
+            build_config: Build config to find build host to terminate.
+        """
+        return
+
+    def __repr__(self) -> str:
+        return f"{type(self)}(NAME={self.NAME()}, build_hosts={self.build_hosts!r} build_hosts_allocated={self.build_hosts_allocated})"
+
+    def __str__(self) -> str:
+        return pprint.pformat(vars(self), width=1, indent=10)
+
+
+class EC2BuildHost(BuildHost):
+    """Class representing an EC2-specific build host instance.
+
+    Attributes:
+        launched_instance_object: Boto instance object associated with the build host.
+    """
+    launched_instance_object: EC2InstanceResource
+
+    def __init__(self, build_config: BuildConfig, inst_obj: EC2InstanceResource, dest_build_dir: str) -> None:
+        super().__init__(build_config=build_config, dest_build_dir=dest_build_dir)
+        self.launched_instance_object = inst_obj
+
+    def __repr__(self) -> str:
+        return f"{type(self)}(build_config={self.build_config!r}, dest_build_dir={self.dest_build_dir}, ip_address={self.ip_address}, launched_instance_object={self.launched_instance_object})"
```

Comment:
```suggestion
        return f"{type(self)}(build_config={self.build_config!r}, dest_build_dir={self.dest_build_dir}, ip_address={self.ip_address}, launched_instance_object={self.launched_instance_object!r})"
```
You probably want to make sure to use the `repr()` of the EC2InstanceResource.